src/osemgrep/language_server/Test_LS_e2e.ml

```diff
@@ -144,7 +121,16 @@ let open_and_write_default_content ?(mode = []) file =
    how the promises resolve, such that the sleeping thread would stay
    sleeping and the entire process would eventually exit.
    So, when running in JSCaml, let's just not use pauses. *)
-let lwt_pause () = if !Common.jsoo then Lwt.return_unit else Lwt.pause ()
+let lwt_pause () =
+  if !Common.jsoo then Lwt.return_unit
+  else
+    (* For some reason, Lwt.pause makes the Sys.cwd change.
```

Comment:
probably because of the bug in chdir_root_path when there is an exn.

Reply:
It seems it's still doing this, even after I got rid of the `chdir_to_repo_root` function. Unsure why.